### PR TITLE
Fix a number of bugs

### PR DESCRIPTION
**Pull request by gcampax**
These are bugs that I found over the course of using dbus-native.
I thought dbus-native was unmantained, but I'm seeing activity recently so I hope these can be merged so I don't need to keep the fork alive.

Each commit is a different bug.

### CODE REVIEW

File: lib/bus.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
+// -*- mode: js; c-basic-offset: 2; js-basic-offset: 2 -*-
+
```
Comment (jrobeson) [inline lib/bus.js]: I don't think we want modelines.

**Comment by sidorares** [comment]
looks good, sorry it was hanging for so long

**Comment by gcampax** [comment]
Oh great, thank you!

**Comment by gcampax** [comment]
I forgot to say, this PR should be a fix for #10, #9 and #143 (but without the tests in #143 )

**Comment by sidorares** [comment]
Can you confirm if it actually fixes issues and I'll close those?

**Comment by gcampax** [comment]
It certainly fixes #10 and #9. I'm not completely sure about #143, but this PR fixes a bug in the area of variant marshalling so it is quite likely it's the same bug.

File: lib/bus.js (hunk 1 above)
Comment (jrobeson) [reply lib/bus.js]: can you remove the modelines?

**Pull request merged**